src/thriveopedia/fossilisation/MuseumCard.cs

```diff
@@ -11,15 +11,30 @@ public class MuseumCard : Button
     [Export]
     public NodePath SpeciesPreviewPath = null!;
 
+    [Export]
+    public NodePath DeleteButtonPath = null!;
+
+    [Export]
+    public NodePath DeleteConfirmationDialogPath = null!;
+
     private Label? speciesNameLabel;
     private TextureRect? speciesPreview;
 
     private Species? savedSpecies;
     private Image? fossilPreviewImage;
 
+    private TextureButton? deleteButton;
```

Comment:
You can use nullable suppression here (as is commonly used elsewhere) to avoid throwing all those SceneTreeAttachRequired thing.